.travis/shellchecker

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env bash
+#-------------------------------------------------------------------------------
+# Copyright (C) 2012-2019 British Crown (Met Office) & Contributors.
+#
+# This file is part of Rose, a framework for meteorological suites.
+#
+# Rose is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Rose is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Rose. If not, see <http://www.gnu.org/licenses/>.
+#-------------------------------------------------------------------------------
+
+# Wrapper for running `shellcheck` over projects.
+
+set -e
+cd "$(dirname "$0")/../"
+
+# find shell files under the specified directory
+find_files () {
+    FPATH="$1"
+    grep --color=never -rl '^#\!.*bash' "${FPATH}" | sed 's/^\.\///' &
+    find "${FPATH}" -name '*.sh' | sed 's/^\.\///'
+}
+
+main () {
+    # parse CLI
+    includes=()
+    excludes=()
+    args=()
+    while [[ "$#" -gt 0 ]]; do
+        if [[ ${1} == '--' ]]; then
+            shift
+            break
+        elif [[ ${1} == "--exclude" ]]; then
+            shift
+            excludes+=("$1")
+        elif [[ ${1} == "-*" ]]; then
+            args+=("$1")
+        else
+            mapfile -t files < <(find_files "${1}")
+            includes=("${includes[@]}" "${files[@]}")
+        fi
+        shift
+    done
+
+    # extract files list
+    files=()
+    for include in "${includes[@]}"; do
+        skip=0
+        for exclude in "${excludes[@]}"; do
+            if [[ "${include}" == "${exclude}"* ]]; then
+                skip=1
+                break
+            fi
+        done
+        if [[ $skip == 0 ]]; then
+            files+=("${include}")
+        fi
+    done
+
+    # sort $files removing duplicates
+    readarray -t files < <(for x in "${files[@]}"; do echo "$x"; done | sort -u)
+
+    # run shellcheck
+    shellcheck "$@" "${files[@]}"
+}
+
+# default configuration for linting cylc
+default () {
+    # run a strict check on all "functional" scripts
+    main bin etc lib/bash sbin usr t/lib/bash \
+        --exclude etc/rose-bash-completion \
```

Comment:
I did have a go at fixing this file but it proved to be flaky. We will need to re-write it with the retirement of `rose suite-run` anyway.

Reply:
I had a couple of goes at this file but I kept breaking the tests. This script will have to be re-written (or at least heavily edited) soon.